appender-core/src/main/java/com/van/logging/aws/S3Configuration.java

```diff
@@ -86,6 +89,13 @@ public void setSseConfiguration(S3SSEConfiguration sseConfiguration) {
         this.sseConfiguration = sseConfiguration;
     }
 
+    public CannedAccessControlList getCannedAcl() {
+        return cannedAcl;
+    }
+
+    public void setCannedAcl(CannedAccessControlList cannedAcl) {
+        this.cannedAcl = cannedAcl;
+    }
```

Comment:
I would add a version to accept a String:
```
/**
 * Sets the canned ACL for S3 to use when storing objects.
 *
 * @param cannedAclValue - the String value of the ACL to set (values for the CannedAccessControlList enum)
 * 
 * @throws IllegalArgumentException - if the cannedAclValue cannot be mapped to a valid CannedAccessControlList enum
 */
public void setCannedAclFromValue(String cannedAclValue) throws IllegalArgumentException {
    this.cannedAcl = CannedAccessControlList.valueOf(cannedAclValue);
}
```
that **appender-log4j** and **appender-log4j2** can call with just the String. 

This will remove the dependency on **aws-java-sdk-s3** by those two modules. Remember: this library can work with Azure and GCP as well. Users of those cloud storage solutions should not need to depend on AWS libraries.

Reply:
Agree, makes sense. Incorporated the needed changes.